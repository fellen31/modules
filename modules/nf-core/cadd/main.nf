process CADD {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container 'docker.io/biocontainers/cadd-scripts:1.7--hdfd78af_0'

    containerOptions {
        (workflow.containerEngine == 'singularity') ?
            "-B ${annotation_dir}:/opt/CADD-scripts-1.7/data/annotations" :
            "-v ${annotation_dir}:/opt/CADD-scripts-1.7/data/annotations"
        }

    input:
    tuple val(meta), path(vcf)
    path(annotation_dir)

    output:
    tuple val(meta), path("*.tsv.gz"), emit: tsv
    path "versions.yml"              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = "1.7" // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
    """
    cadd.sh \\
        -o ${prefix}.tsv.gz \\
        $args \\
        $vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cadd: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = "1.7" // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
    """
    echo echo "" | gzip > ${prefix}.tsv.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cadd: $VERSION
    END_VERSIONS
    """
}
