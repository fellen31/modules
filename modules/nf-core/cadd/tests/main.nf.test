// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test cadd
nextflow_process {

    name "Test Process CADD"
    script "../main.nf"
    process "CADD"

    tag "modules"
    tag "modules_nfcore"
    tag "cadd"

    test("homo_sapiens - vcf - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                 input[0] = [
                 [ id:'test', single_end:false ], // meta map
                 file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gvcf/test.genome.vcf', checkIfExists: true)
                ]
                input[1] = file('/path/to/dir')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
