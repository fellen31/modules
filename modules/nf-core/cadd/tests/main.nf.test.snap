{
    "homo_sapiens - vcf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv.gz:md5,07b334d448bc4bcd9620bb4e8ac11c42"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ef02d93c7627a5a20a25326b5d7ebffc"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv.gz:md5,07b334d448bc4bcd9620bb4e8ac11c42"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ef02d93c7627a5a20a25326b5d7ebffc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-26T11:47:05.583114003"
    }
}